### PR TITLE
mgr/dashboard: fix visibility of pwdExpirationDate field

### PR DESCRIPTION
**Pull request by s0nea**
Show the 'pwdExpirationDate' form field only if SSO is not enabled. Also show a helper message if 'pwdExpirationSpan' is '0' to let the admin know that the password will only expire once.

Screenshot_2020-01-17_10-25-50

Fixes: https://tracker.ceph.com/issues/43523
Signed-off-by: Tatjana Dehler <tdehler@suse.com>

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/src/app/core/auth/user-form/user-form.component.html [hunk 1]
```diff
@@ -103,13 +103,26 @@
...
+             *ngIf="!authStorageService.isSSO()">
           <label class="col-form-label col-sm-3"
                  for="pwdExpirationDate">
             <ng-container i18n>Password expiration date</ng-container>
+            <cd-helper class="text-pre"
+                       *ngIf="pwdExpirationSettings.pwdExpirationSpan == 0">
+              <p>
+                The Dashboard setting defining the expiration interval of
+                passwords is set to <strong>0</strong> currently. It means the
+                user password will only expire once.
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/core/auth/user-form/user-form.component.html]:
```
The Dashboard setting defining the expiration interval of
passwords is currently set to <strong>0</strong>. This means that, 
if a date is set, the user password will only expire once.
```

File: src/pybind/mgr/dashboard/frontend/src/app/core/auth/user-form/user-form.component.html [hunk 2]
```diff
@@ -103,13 +103,26 @@
...
+              </p>
+              <p>
+                Consider to configure the Dashboard setting
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/core/auth/user-form/user-form.component.html]:
```suggestion
                Consider configuring the Dashboard setting
```

File: src/pybind/mgr/dashboard/frontend/src/app/core/auth/user-form/user-form.component.html (hunk 1 above)
Comment (s0nea) [reply src/pybind/mgr/dashboard/frontend/src/app/core/auth/user-form/user-form.component.html]: Thanks! Would you also be fine with:

```
The Dashboard setting defining the expiration interval of
passwords is currently set to <strong>0</strong>. This means
if a date is set, the user password will only expire once.
```

?

Comment (tspmelo) [reply src/pybind/mgr/dashboard/frontend/src/app/core/auth/user-form/user-form.component.html]: Sure.

File: src/pybind/mgr/dashboard/frontend/src/app/core/auth/user-form/user-form.component.html [hunk 3]
```diff
@@ -103,13 +103,26 @@
...
+              <p>
+                The Dashboard setting defining the expiration interval of
+                passwords is currently set to <strong>0</strong>. This means
+                if a date is set, the user password will only expire once.
+              </p>
+              <p>
+                Consider configuring the Dashboard setting
+                <a routerLink="/mgr-modules/edit/dashboard"
+                   class="alert-link">USER_PWD_EXPIRATION_SPAN</a>
+                in order to expire passwords periodically.
```
Comment (callithea) [inline src/pybind/mgr/dashboard/frontend/src/app/core/auth/user-form/user-form.component.html]:
```suggestion
                in order to let passwords expire periodically.
```

**Pull request merged**